lib/jbuilder.rb

```diff
@@ -281,8 +277,16 @@ def target!
 
   private
 
+  def _extract(object, attributes)
+    if ::Hash === object
+      _extract_hash_values(object, attributes)
+    else
+      _extract_method_values(object, attributes)
+    end
+  end
+
   def _extract_hash_values(object, attributes)
-    attributes.each{ |key| _set_value key, _format_keys(object.fetch(key)) }
+    attributes.each{ |key| _set_value key, _format_keys(object[key]) }
```

Comment:
Also using `[]` instead of `fetch`. It's a bit faster.

Reply:
Is the behaviour identical on missing keys?

Reply:
Good question.  `fetch` actually throws if the key doesn't exist, vs returning `nil`. Not clear if this is the _intended_ behaviour or just happenstance. The coding style for the project seems to prefer `fetch`. I git blamed back 10+ years to see if there was any reasoning for this, but saw nothing of note before losing interest.

Mixed feelings on this. The current implementation would result in 500 errors if/when this occurs, which is not something I would expect or desire from our tooling. But given my intent here was simply performance - not a behavioural change - I will switch this back to `fetch`; the savings still fall within the margin of error (according to ips) and not worth the risk and drift from the upstream project.